apps/app/src/test/scala/org/lfdecentralizedtrust/splice/integration/tests/ScanTotalSupplyBigQueryIntegrationTest.scala

```diff
@@ -0,0 +1,509 @@
+
+  // Datastream variant of update_history_creates
+  // contains Amulet, LockedAmulet, UnclaimedReward, &c
+  private val createsSchema = Seq(
+    cc("history_id", INTEGER),
+    cc("row_id", INTEGER),
+    cc("ingested_at", TIMESTAMP),
+    cc("event_id", STRING),
+    cc("update_row_id", INTEGER),
+    cc("contract_id", STRING),
+    cc("created_at", INTEGER),
+    cc("template_id_package_id", STRING),
+    cc("template_id_module_name", STRING),
+    cc("template_id_entity_name", STRING),
+    cc("package_name", STRING),
+    cc("create_arguments", JSON),
+    cc("signatories", JSON, pgIsTextArray = true),
+    cc("observers", JSON, pgIsTextArray = true),
+    cc("contract_key", JSON),
+    cc("record_time", INTEGER),
+    cc("update_id", STRING),
+    cc("domain_id", STRING),
+    cc("migration_id", INTEGER),
+  )
+
+  // Datastream variant of update_history_exercises
+  // contains AmuletRules_Transfer &c
+  private val exercisesSchema = Seq(
+    cc("history_id", INTEGER),
+    cc("row_id", INTEGER),
+    cc("ingested_at", TIMESTAMP),
+    cc("event_id", STRING),
+    cc("update_row_id", INTEGER),
+    cc("child_event_ids", JSON, pgIsTextArray = true),
+    cc("choice", STRING),
+    cc("template_id_package_id", STRING),
+    cc("template_id_module_name", STRING),
+    cc("template_id_entity_name", STRING),
+    cc("contract_id", STRING),
+    cc("consuming", BOOLEAN),
+    cc("argument", JSON),
+    cc("result", JSON),
+    cc("package_name", STRING),
+    cc("interface_id_package_id", STRING),
+    cc("interface_id_module_name", STRING),
+    cc("interface_id_entity_name", STRING),
+    cc("acting_parties", JSON, pgIsTextArray = true),
+    cc("record_time", INTEGER),
+    cc("update_id", STRING),
+    cc("domain_id", STRING),
+    cc("migration_id", INTEGER),
+  )
+
+  private type ConvertibleSchema = Seq[ConvertibleColumn]
+
+  private def bqSchema(schema: ConvertibleSchema): Schema =
+    Schema.of(schema.map(_.bqSchemaField)*)
+
+  // create empty tables in BigQuery that match the schema inferred by Datastream,
+  // less the datastream_metadata column (which we don't use)
+  private def createEmptyTables(): Unit = {
+    // row_id is primary key but this is not currently enforced even in actual
+    // deployment
+    createTable(createsBqTableName, createsSchema)
+    createTable(exercisesBqTableName, exercisesSchema)
+  }
+
+  private def createTable(tableName: String, schema: ConvertibleSchema): Unit = {
+    val tableId = TableId.of(datasetName, tableName)
+    val tableDefinition = bq.StandardTableDefinition of bqSchema(schema)
+    val tableInfo = bq.TableInfo.of(tableId, tableDefinition)
+    bigquery.create(tableInfo)
+  }
+
+  /** Creates test data with known amounts for all metrics
+    */
+  private def createTestData(aliceParty: PartyId, bobParty: PartyId)(implicit
+      env: FixtureParam
+  ): Unit = {
+    // TODO (#1713) use a realistic minting method; best not to support tap in the SQL
```

Comment:
I think this is fine for now to leave it as a TODO, thanks.

Reply:
Depending on if this is needed for regression test on correct totals, mints and burns, might be a good idea to quickly follow up with that.

Reply:
👍